adcm_aio_client/core/objects/cm.py

```diff
@@ -480,15 +486,27 @@ def object(self: Self) -> Object:
     def action(self: Self) -> Action:
         return self._construct(what=Action, from_data=self._data["action"])
 
-    async def wait(self: Self, status_predicate: Callable[[], bool], timeout: int = 30, poll: int = 5) -> None:
-        if self._data["status"] not in (JobStatus.RUNNING, JobStatus.CREATED):
-            return
+    async def wait(
+        self: Self, timeout: int | None, poll_interval: int = 10, exit_condition: Optional[Callable[[], bool]] = None
+    ) -> Self:
+        if self._data["status"] not in ("running", "created"):
+            return self
```

Comment:
On one hand, optimization is understandable, but on the other one it contradicts `exit_condition` which may wait until `running` is reached or whatever non-status related stuff.